internal/docstore/firedocstore/fs.go

```diff
@@ -557,24 +590,34 @@ func (c *collection) commit(ctx context.Context, ws []*pb.Write) ([]*pb.WriteRes
 	return res.WriteResults, nil
 }
 
-// docName returns the name of the document. This is the value of
-// the field called c.nameField, and it must be a string.
-// If the field doesn't exist, docName returns the empty string, rather
-// than an error. This is to support the Create action, which can
-// create new document names.
-func (c *collection) docName(doc driver.Document) (string, error) {
-	n, err := doc.GetField(c.nameField)
-	if err != nil {
-		// Return missing field as empty string.
-		return "", nil
+// docName returns the name of the document. This is either the value of the field
+// called c.nameField, or the result of calling c.nameFunc. It must be a
+// string.
+// docName returns an error if NameField isn't present, or NameFunc returns "".
```

Comment:
Should this be an "and" in between? It returns error when neither name field nor name func works right?

Reply:
I reworded to clarify.